crates/solver/src/liquidity/zeroex.rs

```diff
@@ -86,28 +136,7 @@ impl LiquidityCollecting for ZeroExLiquidity {
         pairs: HashSet<TokenPair>,
         _block: Block,
     ) -> Result<Vec<Liquidity>> {
-        let queries = &[
-            // orders fillable by anyone
-            OrdersQuery::default(),
-            // orders fillable only by our settlement contract
-            OrdersQuery {
-                sender: Some(self.gpv2.address()),
-                ..Default::default()
-            },
-        ];
-
-        let zeroex_orders_results =
-            futures::future::join_all(queries.iter().map(|query| self.api.get_orders(query))).await;
-        let zeroex_orders = zeroex_orders_results
-            .into_iter()
-            .flat_map(|result| match result {
-                Ok(order_record_vec) => order_record_vec,
-                Err(err) => {
-                    tracing::warn!("ZeroExResponse error during liqudity fetching: {}", err);
-                    vec![]
-                }
-            });
-
+        let zeroex_orders = self.orderbook_cache.read().await.clone();
```

Comment:
Neat. After looking into liquidity fetching performance over the last couple of days I came to the conclusion that all individual pieces of liquidity should be `Arc`ed internally to make these operations very cheap. When we actually use the real orderbook (not the fake one for testing) the cloning alone will use a significant amount of time.
This does not have to happen in this PR (to not delay further experiments) but overall I think that way we can accomplish an almost 0 overhead liquidity fetching implementation and revert back to running a single driver.